lib/cli/src/cli.rs

```diff
@@ -344,14 +344,34 @@ fn test_split_version() {
     );
 }
 
+#[derive(Debug)]
+pub(crate) enum SplitVersionError {
+    InvalidCommandName(anyhow::Error, Option<anyhow::Error>),
```

Comment:
`anyhow::Error` is designed to be a top level error type.

Nesting it within other error types doesn't really make sense.

You can either implement a concrete enum with variants that hold the original upstream errors, 
or you can have custom error(s) for specific conditiions that are then wrapped in anyhow and can be checked via downcasting.

But embedding `anhow::Error` doesn't make sense.